pytest 9 incompatible with fixtures

spyder defines some test fixtures via pytest's `FixtureDef` at
https://github.com/spyder-ide/spyder/blob/7c17fc521334bb154d3cd1badd6b85c5a1f173a6/spyder/api/plugins/tests.py#L128

```
       request._fixturemanager._arg2fixturedefs[fixture_name] = [
            FixtureDef(
                argname=fixture_name,
                func=register_plugin_factory(plugin_cls),
                scope="session",
                fixturemanager=request._fixturemanager,
                baseid=request.node.nodeid,
                params=None,
            )
        ]
```

pytest 9 has changed the signature of `FixtureDef` in https://github.com/pytest-dev/pytest/pull/11794

In particular `fixturemanager` has been replaced with `config: Config`.

This means that tests using these fixtures, such as `test_connection_dialog_remembers_input_with_ssh_passphrase` from spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py, fail with pytest 9 with the error
```
E           TypeError: FixtureDef.__init__() got an unexpected keyword argument 'fixturemanager'

spyder/api/plugins/tests.py:124: TypeError
```

--- a/spyder/api/plugins/tests.py
+++ b/spyder/api/plugins/tests.py
@@ -9,6 +9,7 @@ from __future__ import annotations
 
 # Standard library imports
 import gc
+import inspect
 import sys
 from typing import TYPE_CHECKING
 
@@ -32,6 +33,38 @@ __all__ = [
     "plugins_cls",
     "register_fixture",
 ]
+
+
+_FIXTUREDEF_PARAMETERS = inspect.signature(FixtureDef).parameters
+
+
+def _create_fixturedef(
+    request: SubRequest,
+    fixture_name: str,
+    fixture_func,
+) -> FixtureDef:
+    """Create a pytest FixtureDef compatible with the installed pytest."""
+    kwargs = {
+        "argname": fixture_name,
+        "func": fixture_func,
+        "scope": "session",
+        "baseid": request.node.nodeid,
+        "params": None,
+    }
+
+    if "fixturemanager" in _FIXTUREDEF_PARAMETERS:
+        kwargs["fixturemanager"] = request._fixturemanager
+
+    if "config" in _FIXTUREDEF_PARAMETERS:
+        kwargs["config"] = request.config
+
+    if "ids" in _FIXTUREDEF_PARAMETERS:
+        kwargs["ids"] = None
+
+    if "_ispytest" in _FIXTUREDEF_PARAMETERS:
+        kwargs["_ispytest"] = True
+
+    return FixtureDef(**kwargs)
 
 
 class MainWindowMock(QMainWindow):
@@ -125,12 +158,9 @@ def register_fixture(request: SubRequest, plugins_cls) -> None:
             return register_plugin
 
         request._fixturemanager._arg2fixturedefs[fixture_name] = [
-            FixtureDef(
-                argname=fixture_name,
-                func=register_plugin_factory(plugin_cls),
-                scope="session",
-                fixturemanager=request._fixturemanager,
-                baseid=request.node.nodeid,
-                params=None,
+            _create_fixturedef(
+                request,
+                fixture_name,
+                register_plugin_factory(plugin_cls),
             )
         ]

--- a/spyder/api/tests/test_plugin_fixtures.py
+++ b/spyder/api/tests/test_plugin_fixtures.py
@@ -1,0 +1,98 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2026- Spyder Project Contributors
+#
+# Released under the terms of the MIT License
+# (see LICENSE.txt in the project root directory for details)
+# -----------------------------------------------------------------------------
+
+"""Tests for plugin testing helpers."""
+
+from types import SimpleNamespace
+
+from spyder.api.plugins import tests as plugin_tests
+
+
+def test_create_fixturedef_with_fixturemanager(monkeypatch):
+    """Test creating a FixtureDef for pytest versions before 9."""
+    kwargs = {}
+
+    class DummyFixtureDef:
+        def __init__(self, **fixture_kwargs):
+            kwargs.update(fixture_kwargs)
+
+    request = SimpleNamespace(
+        config=object(),
+        node=SimpleNamespace(nodeid="nodeid"),
+        _fixturemanager=object(),
+    )
+    fixture_func = object()
+
+    monkeypatch.setattr(plugin_tests, "FixtureDef", DummyFixtureDef)
+    monkeypatch.setattr(
+        plugin_tests,
+        "_FIXTUREDEF_PARAMETERS",
+        {
+            "argname": None,
+            "func": None,
+            "scope": None,
+            "fixturemanager": None,
+            "baseid": None,
+            "params": None,
+        },
+    )
+
+    plugin_tests._create_fixturedef(request, "plugin_fixture", fixture_func)
+
+    assert kwargs == {
+        "argname": "plugin_fixture",
+        "func": fixture_func,
+        "scope": "session",
+        "fixturemanager": request._fixturemanager,
+        "baseid": "nodeid",
+        "params": None,
+    }
+
+
+def test_create_fixturedef_with_config(monkeypatch):
+    """Test creating a FixtureDef for pytest 9 and later."""
+    kwargs = {}
+
+    class DummyFixtureDef:
+        def __init__(self, **fixture_kwargs):
+            kwargs.update(fixture_kwargs)
+
+    request = SimpleNamespace(
+        config=object(),
+        node=SimpleNamespace(nodeid="nodeid"),
+        _fixturemanager=object(),
+    )
+    fixture_func = object()
+
+    monkeypatch.setattr(plugin_tests, "FixtureDef", DummyFixtureDef)
+    monkeypatch.setattr(
+        plugin_tests,
+        "_FIXTUREDEF_PARAMETERS",
+        {
+            "config": None,
+            "baseid": None,
+            "argname": None,
+            "func": None,
+            "scope": None,
+            "params": None,
+            "ids": None,
+            "_ispytest": None,
+        },
+    )
+
+    plugin_tests._create_fixturedef(request, "plugin_fixture", fixture_func)
+
+    assert kwargs == {
+        "config": request.config,
+        "baseid": "nodeid",
+        "argname": "plugin_fixture",
+        "func": fixture_func,
+        "scope": "session",
+        "params": None,
+        "ids": None,
+        "_ispytest": True,
+    }

